packages/editor/src/components/post-title/post-title-rich.js

```diff
@@ -0,0 +1,170 @@
+/**
+ * WordPress dependencies
+ */
+import { forwardRef, useState } from '@wordpress/element';
+import { ENTER } from '@wordpress/keycodes';
+import { useDispatch } from '@wordpress/data';
+import { pasteHandler } from '@wordpress/blocks';
+import { store as blockEditorStore } from '@wordpress/block-editor';
+import {
+	__unstableUseRichText as useRichText,
+	create,
+	toHTMLString,
+	insert,
+} from '@wordpress/rich-text';
+import { useMergeRefs } from '@wordpress/compose';
+import { __unstableStripHTML as stripHTML } from '@wordpress/dom';
+
+/**
+ * Internal dependencies
+ */
+import PostTypeSupportCheck from '../post-type-support-check';
+
+const PostTitleRich = forwardRef(
+	(
+		{ title, placeholder, onChange, onUpdate, className, setIsSelected },
+		ref
+	) => {
+		const [ selection, setSelection ] = useState( {} );
+
+		const { clearSelectedBlock, insertBlocks, insertDefaultBlock } =
+			useDispatch( blockEditorStore );
+
+		function onInsertBlockAfter( blocks ) {
+			insertBlocks( blocks, 0 );
+		}
+
+		function onSelect() {
+			setIsSelected( true );
+			clearSelectedBlock();
+		}
+
+		function onUnselect() {
+			setIsSelected( false );
+			setSelection( {} );
+		}
+
+		function onEnterPress() {
+			insertDefaultBlock( undefined, undefined, 0 );
+		}
+
+		function onKeyDown( event ) {
+			if ( event.keyCode === ENTER ) {
+				event.preventDefault();
+				onEnterPress();
+			}
+		}
+
+		function onPaste( event ) {
+			const clipboardData = event.clipboardData;
+
+			let plainText = '';
+			let html = '';
+
+			// IE11 only supports `Text` as an argument for `getData` and will
+			// otherwise throw an invalid argument error, so we try the standard
+			// arguments first, then fallback to `Text` if they fail.
+			try {
+				plainText = clipboardData.getData( 'text/plain' );
+				html = clipboardData.getData( 'text/html' );
+			} catch ( error1 ) {
+				try {
+					html = clipboardData.getData( 'Text' );
+				} catch ( error2 ) {
+					// Some browsers like UC Browser paste plain text by default and
+					// don't support clipboardData at all, so allow default
+					// behaviour.
+					return;
+				}
+			}
+
+			// Allows us to ask for this information when we get a report.
+			window.console.log( 'Received HTML:\n\n', html );
```

Comment:
Do we still need these lines?

Reply:
It was there originally so I think we leave it in.